Kinvey.Core/Store/FindRequest.cs

```diff
@@ -78,7 +78,40 @@ public override async Task<List<T>> ExecuteAsync()
 					}
 					break;
 
-				default:
+                case ReadPolicy.NETWORK_OTHERWISE_LOCAL:
+                    // auto
+
+                    KinveyException networkKinveyException = null;
+                    try
+                    {
+                        // first, perform a network request
+                        var networkResult = await PerformNetworkFind();
+                        if (networkResult.IsDeltaFetched)
+                        {
+                            listResult = PerformLocalFind();
```

Comment:
What is the reasoning here for performing a local find?

Reply:
also, why only when we are performing delta set fetches?

Reply:
@vinaygahlawat @heyzooi 
Before answering on your questions, I want to say that the functionality is not new for us.
DataStoreCache is already using this approach. If you have a look at a couple of lines in top, you can find the same
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/FindRequest.cs#L73

About your questions. Answers on them are in the "PerformNetworkFind()" method.
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L192

The idea is when the delta set is turn on, we need to make a delta set request (
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L212)
which returns the "DeltaSetResponse"(https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Model/DeltaSetResponse.cs#L24)
Further we are handling the response deleting or creating/updating received entities
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L268
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L271
Summing up. As a result, what do we have?
We have the network response which contains some pieces of data. They are removed or created/updated entities. We can not return it to user since it is only some parts of all data.
And we have actual local cache data which contain all entities. Thus we need to use the "PerformLocalFind()".

If the delta set is turn off, we will make get request.
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L371
It will return all actual entities. After this we will rewrite the local cache
(https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/ReadRequest.cs#L373)
Ultimately what do we have? We have actual data from the network response and in the local cache.
That is why we are calling
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/FindRequest.cs#L95
We could call the “PerformLocalFind()” here too. In this case we don’t need to have this “if”
https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2849-Create_AUTO_DataStore_type/Kinvey.Core/Store/FindRequest.cs#L89.
But it is extra request to the local DB. It makes sense to use existing data in the “networkResult” object.

Reply:
@romanogolikhin thanks for the explanation! it's making more sense in my head now.